### DISCUSSION TITLE
webpack-dev-server v5 seems to require `'unsafe-eval'` in CSP

### DISCUSSION THREAD

**Discussion by G-Rath** [webpack/webpack#18073] [Q&A] [answered]
I'm opening this as a discussion first to get input as this could either be a bug or documentation issue with `webpack-dev-server`.

In a nutshell I've updated a Rails application generated using [this template](https://github.com/ackama/rails-template) which defines a [CSP](https://github.com/ackama/rails-template/blob/main/variants/backend-base/config/initializers/content_security_policy.rb) and includes some [extras for `webpack-dev-server`](https://github.com/ackama/rails-template/blob/main/variants/backend-base/config/initializers/content_security_policy.rb#L94); after upgrading to v5 however, I'm now getting an error about eval'd JS:

<img width="1005" alt="Screenshot 2024-02-13 090922" src="https://github.com/webpack/webpack/assets/3151613/674299e8-8c6b-4feb-aff1-3d8915ae0dff">

Apparently this is happening here:

<img width="597" alt="image" src="https://github.com/webpack/webpack/assets/3151613/998f1bb5-9dfc-451d-bb2d-865230ad0136">

Which is from node_modules/webpack-dev-server/client/modules/logger/index.js.

---

I've had a look around and can't find anything indicating what has actually changed in v5 to cause this (my assumption is maybe a different build config, but could be something else, and this might just be the first of a number of these calls 🤷), but ideally it would be good if we could somehow avoid having to change the CSP since we want it to match as closely to production as possible for debugging and in particular enabling `'unsafe-eval'` is a big one (I have tried looking into using a nonce here, but couldn't get webpack to do so?)

I'm happy to help out with this, but need some pointers from the webpack team to continue :)

**Comment by alexander-akait** [comment, marked as answer]
Yes, I see, let's fix it, I will see today

**Reply by sfc-gh-jforrest** [reply]
Did this end up getting fixed? If so, are you able to share which version it is fixed in? Thanks!

**Reply by arraydude** [reply]
Hi @alexander-akait , I'm also interested on this, do you have any news ?

**Reply by alexander-akait** [reply]
Already fixed, just check your version

**Reply by arraydude** [reply]
Thanks! @alexander-akait which version has this fixed?

**Reply by alexander-akait** [reply]
https://github.com/webpack/webpack-dev-server/releases/tag/v5.0.1
